packages/ui/src/components/va-backtop/VaBacktop.vue

```diff
@@ -65,27 +71,41 @@ export default defineComponent({
 
       scrolled.value = true
 
+      if (targetElement instanceof Window) {
+        window.scrollTo({
+          top: 0,
+          behavior: 'smooth',
+        })
+
+        return
+      }
+
       interval.value = window.setInterval(() => {
-        if (targetElement.scrollTop === 0) {
-          clearInterval(interval.value)
-          scrolled.value = false
-        } else {
-          const next = Math.floor(targetElement.scrollTop - props.speed)
-          targetElement.scrollTo(0, next)
+        if (targetElement instanceof Element) {
+          if (targetElement.scrollTop === 0) {
+            clearInterval(interval.value)
+            scrolled.value = false
+          } else {
+            const next = Math.floor(targetElement.scrollTop - props.speed)
+            targetElement.scrollTo(0, next)
+          }
         }
       }, 15)
     }
 
     const handleScroll = () => {
-      visible.value = targetElement.scrollTop > props.visibilityHeight
+      const targetScrollValue = targetElement instanceof Window
+        ? targetElement.scrollY
+        : targetElement.scrollTop
+      visible.value = targetScrollValue > props.visibilityHeight
     }
 
     onMounted(() => {
       targetElement = getTargetElement()
-      targetElement?.addEventListener('scroll', handleScroll)
+      targetElement.addEventListener('scroll', handleScroll, true)
```

Comment:
Does it prevent to capture scroll event from children?

Reply:
One of the reasons, the second is that without capture scroll event listener not working properly for window.